### PR TITLE
Updating pylint, adding black, adding GitHub ci

### PR DESCRIPTION
**Pull request by devinmatte**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[tool.black]
+# default is 88
+line-length = 120
```
Comment (mxmeinhold) [inline pyproject.toml:3]: Can we drop this back to 88? 120 I think is good for a linter, but too much for a formatter.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+skip-string-normalization = true
```
Comment (mxmeinhold) [inline pyproject.toml:4]: Oh good this is my primary complaint with black. We'll use pylint_quotes instead?

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+black~=20.8b1
```
Comment (mxmeinhold) [inline requirements.txt:1]: Would it be worthwile to split the lint dependencies separately? That at least should be our practice for csh-ldap to limit dependency conflicts.

**Review by mxmeinhold — COMMENTED** [review]
@devinmatte Could you split this into several commits? Mostly just to separate the black changes. Not too important, just convenience.

I'm normally not a fan of black, mostly because of its stance on strings. What is it really getting us here that pylint wasn't? Slightly more consistent formatting?

File: pyproject.toml (hunk 1 above)
Comment (devinmatte) [reply pyproject.toml:3]: It will autoformat every line to 88 though. I fucking hate 88 length limits

File: pyproject.toml (hunk 2 above)
Comment (devinmatte) [reply pyproject.toml:4]: Yep pylint_quotes will handle those issues

File: requirements.txt (hunk 3 above)
Comment (devinmatte) [reply requirements.txt:1]: I don't really see that being useful, if anything it would be annoying especially in ide's and editors like vscode

**Comment by devinmatte** [comment]
Overall adding adding black allows quick and easy cleanup of files to conform to style.
I _could_ separate out, but that seems like extra work for very little benefit. It's style fixes and fixing the 3 new pylint errors that came up, none of which changes any functionality

File: pyproject.toml (hunk 1 above)
Comment (mxmeinhold) [reply pyproject.toml:3]: My complaint is that it's reformatting things to 120 that had been manually formatted to be shorter and more readable.

Comment (devinmatte) [reply pyproject.toml:3]: Alright can we compromise at 100?

**Review by mxmeinhold — APPROVED** [review]
I don't really like black but I'll approve this

File: requirements.txt (hunk 3 above)
Comment (mxmeinhold) [reply requirements.txt:1]: Worth keeping in mind for csh_ldap.

**Pull request merged**